src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -558,19 +558,32 @@ private void AttributeValue(HtmlSymbolType quote)
 
             if (At(HtmlSymbolType.Transition))
             {
-                var valueStart = CurrentLocation;
-                PutCurrentBack();
+                if (NextIs(HtmlSymbolType.Transition))
+                {
+                    // Render a single "@" in place of "@@".
+                    Output(SpanKind.Markup);
+                    AcceptAndMoveNext();
+                    Span.CodeGenerator = SpanCodeGenerator.Null;
+                    Output(SpanKind.Markup, AcceptedCharacters.None);
+                    AcceptAndMoveNext();
+                    Output(SpanKind.Markup, AcceptedCharacters.None);
```

Comment:
Based on offline conversation, you've addressed both of my bullets with this comment and your most recent changes.